src/main/java/dal/fatih/todoly/Main.java

```diff
@@ -43,49 +58,64 @@ public static Task handleCreateTask() {
     }
 
     public static void listAllTasks() {
-        if (tasks.isEmpty()) {
-            System.out.println("Task list is empty");
-        } else {
-            for (Map.Entry task : tasks.entrySet()) {
+        try {
+            Set set = tasks.entrySet();
+            Iterator iterator = set.iterator();
+            while (iterator.hasNext()) {
+                Map.Entry task = (Map.Entry) iterator.next();
+
                 System.out.println(task.getValue());
                 System.out.println("----------------------------------");
             }
+        } catch (Exception e) {
+            System.out.println(e);
         }
     }
 
     public static void showTaskDetails() {
-        System.out.print("Task Id :");
-        String taskId = scn.nextLine();
-        Task task = tasks.get(taskId);
-        if (task != null) {
-            System.out.println(tasks.get(taskId));
-        } else {
-            System.out.println("Task not found");
+        try {

```

Comment:
try/catch doesn't seem to be needed. `showTaskDetails ` method should not be changed at all in this PR